snap/validate.go

```diff
@@ -666,6 +665,15 @@ func ValidateApp(app *AppInfo) error {
 		}
 	}
 
+	// ActivatesOn is a list of slot names that use the "dbus" type
+	for _, slotName := range app.ActivatesOn {
```

Comment:
shouldn't we check somehow that the app is a daemon: dbus or is that orthogonal?